.github/workflows/kakarot_rpc.yml

```diff
@@ -1,59 +1,97 @@
 ---
 name: RPC
 
-on:
-  release:
-    types: [published]
+on: [pull_request]
 
 env:
   REGISTRY_IMAGE: ghcr.io/${{ github.repository }}/node
+  DOCKER_USERNAME: ${{ github.actor }}
 
 jobs:
+  prepare:
+    runs-on: ubuntu-latest-16-cores
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+      # Install Foundry
+      - name: install foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      # Install Python
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v3
+        with:
+          path: ~/.local
+          key: poetry-${{ runner.os }}
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Setup the Kakarot submodule
+        run: make setup
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: mkdir -p ./artifacts/contracts && mv ./lib/kakarot/build/* ./artifacts/contracts
```

Comment:
it's this redundant with extract_abi.sh script

Reply:
yes it was, removed